tapeagents/nodes.py

```diff
@@ -52,7 +52,16 @@ def prepare_tape(self, tape: Tape) -> Tape:
         return tape.model_copy(update=dict(steps=steps_without_control_flow))
 
     def make_llm_output(self, agent: Any, tape: Tape, index: int) -> LLMOutput:
-        return LLMOutput(role="assistant", content=tape.steps[index].llm_view())
+        steps = []
+        i = index
+        first_prompt_id = tape.steps[i].metadata.prompt_id
+        while i < len(tape) and tape.steps[i].metadata.prompt_id == first_prompt_id:
+            if not isinstance(tape.steps[i], SetNextNode):
+                steps.append(tape.steps[i])
+            i += 1
+
+        content = [step.llm_dict() for step in steps] if len(steps) > 1 else steps[0].llm_dict()
```

Comment:
I think we need `steps[-1].llm_dict()` here instead

Reply:
Some of our agents produce more than 1 step per call. It is correctly handled by tracking the same prompt_id.